### PR TITLE
Show notices as diagnostics by default

### PR DESCRIPTION
**Pull request by jim-portegies**
Best effect with the changes currently in [this PR](https://github.com/impermeable/coq-waterproof/pull/84)

### CODE REVIEW

File: src/lsp-client/clientTypes.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ export interface CoqLspServerConfig {
     client_version: string;
     eager_diagnostics: boolean;
     goal_after_tactic: boolean;
-    show_coq_info_messages: boolean;
+    show_waterproof_info_messages: boolean;
```
Comment (DikieDick) [inline src/lsp-client/clientTypes.ts:122]:
```suggestion
    show_coq_info_messages: boolean;
```

We can't rename these as they are passed to the `coq-lsp` instance and parsed there. So internally we have to keep the 'coq' name, while for the external/user facing side 'waterproof' is fine.

File: src/lsp-client/clientTypes.ts [hunk 2]
```diff
@@ -139,7 +139,7 @@ export namespace CoqLspServerConfig {
             client_version: client_version,
             eager_diagnostics: wsConfig.eager_diagnostics,
             goal_after_tactic: wsConfig.goal_after_tactic,
-            show_coq_info_messages: wsConfig.show_coq_info_messages,
+            show_waterproof_info_messages: wsConfig.show_waterproof_info_messages,
```
Comment (DikieDick) [inline src/lsp-client/clientTypes.ts:142]:
```suggestion
            show_coq_info_messages: wsConfig.show_waterproof_info_messages,
```

See comment above.

**Review by DikieDick — CHANGES_REQUESTED** [review]

File: src/lsp-client/clientTypes.ts (hunk 2 above)
Comment (jim-portegies) [reply src/lsp-client/clientTypes.ts:142]: Thanks! Apparently I didn't know how this worked.

**Review by DikieDick — COMMENTED** [review]
Looks good to me.

**Review by DikieDick — APPROVED** [review]

**Pull request merged**